resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesTaskSetManager.scala

```diff
@@ -52,12 +56,28 @@ private[spark] class KubernetesTaskSetManager(
           if (pendingTasksClusterNodeIP.nonEmpty) {
             logDebug(s"Got preferred task list $pendingTasksClusterNodeIP for executor host " +
               s"$executorIP using cluster node IP $clusterNodeIP")
+            pendingTasksClusterNodeIP
+          } else {
+            val clusterNodeFullName = inetAddressUtil.getFullHostName(clusterNodeIP)
+            val pendingTasksClusterNodeFullName = super.getPendingTasksForHost(clusterNodeFullName)
+            if (pendingTasksClusterNodeFullName.nonEmpty) {
+              logDebug(s"Got preferred task list $pendingTasksClusterNodeFullName " +
+                s"for executor host $executorIP using cluster node full name $clusterNodeFullName")
+            }
+            pendingTasksClusterNodeFullName
           }
-          pendingTasksClusterNodeIP
         }
       } else {
         pendingTasksExecutorIP  // Empty
       }
     }
   }
 }
+
+// To support mocks in unit tests.
+private[kubernetes] class InetAddressUtil {
+
+  def getFullHostName(ipAddress: String): String = {
+    InetAddress.getByName(ipAddress).getCanonicalHostName
```

Comment:
does this do a network call?  how long does the call take?  I'm wondering if it makes sense to cache these for a period of time